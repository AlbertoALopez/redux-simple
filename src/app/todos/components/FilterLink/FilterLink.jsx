import React from 'react';

const FilterLink = ({
    filter,
    currentFilter,
    children,
    onClick,
}) => {
    if (filter === currentFilter) {
        return <span>{ children }</span>;
    }
    return (
        <a
            href="#"
            onClick={(e) => {
                e.preventDefault(e);
                onClick(filter);
            }}
        >
            {children}
        </a>
    );
};

export default FilterLink;
