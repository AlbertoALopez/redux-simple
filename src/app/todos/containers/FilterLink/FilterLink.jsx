import { connect } from 'react-redux';
import { setVisibilityFilter } from '../../actions/actions.js';
import Link from '../../components/Link/Link.jsx';


const mapStatetoProps = (state, ownProps) => {
    return {
        active: ownProps.filter === state.visibilityFilter,
    };
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        onClick: () => {
            dispatch(setVisibilityFilter(ownProps.filter));
        },
    };
};

const FilterLink = connect(
    mapStatetoProps,
    mapDispatchToProps,
)(Link);

export default FilterLink;
