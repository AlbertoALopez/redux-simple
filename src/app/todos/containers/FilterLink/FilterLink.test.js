import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import FilterLink from './FilterLink.jsx';


function onFilterClick() {
    console.log('click');
}

test('FilterLink should render', () => {
    const component = renderer.create(
        <FilterLink
            filter="SHOW_ALL"
            currentFilter="SHOW_ALL"
            onClick={onFilterClick}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Filterlink should render a span when the current filter is selected', () => {
    const wrapper = mount(
        <FilterLink
            filter="SHOW_ALL"
            currentFilter="SHOW_ALL"
            onClick={onFilterClick}
        />
    );

    expect(wrapper.find('span').length).toBe(1);
});

test('FilterLink should render an anchor tag when current filter is not selected', () => {
    const wrapper = mount(
        <FilterLink
            filter='SHOW_ALL'
            currentFilter="SHOW_COMPLETED"
            onClick={onFilterClick}
        />
    );

    expect(wrapper.find('a').length).toBe(1);
});
