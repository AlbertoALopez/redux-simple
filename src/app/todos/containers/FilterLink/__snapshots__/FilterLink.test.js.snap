exports[`test FilterLink should render 1`] = `<span />`;
